spark-jobs/src/main/scala/filodb/cardbuster/PerShardCardinalityBuster.scala

```diff
@@ -100,5 +100,6 @@ class PerShardCardinalityBuster(dsSettings: DownsamplerSettings,
     }
     BusterContext.log.info(s"Finished deleting keys from shard shard=$shard " +
       s"numDeleted=$numDeleted numCouldNotDelete=$numCouldNotDelete")
```

Comment:
``numCouldNotDelete`` doesn't change. Is it expected to be ``candidateKeys.size - numDeleted``? I think we can just eliminate ``numCouldNotDelete`` and print ``candidateKeys.size`` and ``numDeleted`` to logs. 
Since ``candidateKeys`` is an iterator, we can simply increment the count in the filter.

Reply:
numCouldNotDelete can change when we are not simulating. What are we trying to optimize here ?

Reply:
I see numCouldNotDelete is what had exception when deletion, I was talking about also printing the number of ``candidateKeys``. As of now the simulation run doesn't tell me how many candidate keys matched  **before** filtering.